graphql_compiler/compiler/ir_sanity_checks.py

```diff
@@ -139,34 +176,33 @@ def _sanity_check_mark_location_preceding_optional_traverse(ir_blocks):
 
 def _sanity_check_every_location_is_marked(ir_blocks):
     """Ensure that every new location is marked with a MarkLocation block."""
-    # Exactly one MarkLocation block is found between a QueryRoot / Traverse / Recurse block,
-    # and the first subsequent Traverse, Recurse, Backtrack or ConstructResult block.
+    # Exactly one MarkLocation block is found between any block that starts an interval of blocks
```

Comment:
Great question! It started with Gremlin (the first language that we built a backend for), where marking a location does not commute with filtering or other operations like this, so you have to mark after you've pruned everything you wanted pruned. Otherwise, it was kind of a toss-up between marking before and after, so we chose to mark after since it required less code.